api/http/proxy/transport.go

```diff
@@ -258,7 +258,7 @@ func (p *proxyTransport) proxyImageRequest(request *http.Request) (*http.Respons
 	case "/images/create":
 		return p.replaceRegistryAuthenticationHeader(request)
 	default:
-		if match, _ := path.Match("/images/*/push", requestPath); match {
+		if match, _ := path.Match("/images/*/*/push", requestPath); match {
```

Comment:
Actually, I think that we should use the following condition here:

```
if path.Base(requestPath) == "push" && request.Method == http.MethodPost {
  return p.replaceRegistryAuthenticationHeader(request)
}
```

Could you update it?